src/cmd/ksh93/tests/builtins.sh

```diff
@@ -686,5 +686,10 @@ PATH=$tmp:$PATH $SHELL <<-\EOF || err_exit "'whence' gets wrong path on init"
 	[[ -x $wc ]]
 EOF
 
+# ======
+# `builtin -d` should not delete special builtins
+builtin -d export 2> /dev/null
+whence -q export || err_exit '`builtin -d` deletes special builtins'
+
```

Comment:
Deleting `export` in the main shell environment will affect subsequent tests that may be added. I would prefer it if the test were non-destructive by executing `builtin -d` plus the check in a subshell:

```sh
(builtin -d export 2> /dev/null
whence -q export) || err_exit '`builtin -d` deletes special builtins'
```

Reply:
Thanks for pointing that out, it is now fixed.

Reply:
…I just thought of another thing, sorry. An external `export` command may exist on some systems. So to avoid a false positive, we need to disable the PATH search in the subshell.

(builtin -d export 2> /dev/null
PATH=/dev/null
whence -q export) || err_exit '`builtin -d` deletes special builtins'

Reply:
That is now fix as well.